File: Modulo04/AlunoProfessor-CSharp/Pessoa.cs
using System;
public class Pessoa {

    private String nome;
    private int idade;

    private static int contador = 0;

    public Pessoa(String nome, int idade) {        
        this.nome = nome;
        this.idade = idade;
        incrementa();
        Console.WriteLine("Construtor: " + this.nome);
    }

    ~Pessoa() {
        Pessoa.decrementa();
        Console.WriteLine("Destrutor: " + this.nome);
    }

    public int getIdade() {
        return this.idade;
    }
    
    public void setIdade(int idade) {
        this.idade = idade;
    }
    
    public String getNome() {
        return this.nome;
    }
    
    public void setNome(String nome) {
        this.nome = nome;
    }
    
    public virtual void imprime() {
        Console.WriteLine("Nome: " + this.nome);
        Console.WriteLine("Idade: " + this.idade);
    }
    
    public static int getContador() {
        return Pessoa.contador;
    }

    private static void decrementa() {
        Pessoa.contador -= 1;
    }

    private static void incrementa(){
        Pessoa.contador += 1;
    }
}

File: Modulo04/AlunoProfessor-Cpp/Pessoa.cpp

#include "Pessoa.h"

int Pessoa::contador = 0;

Pessoa::Pessoa() {
    incrementa();
    cout << "Construtor Pessoa()" << endl;
}

Pessoa::Pessoa(string nome, int idade) :
nome(nome), idade(idade) {
    incrementa();
    cout << "Construtor: " << this->nome << endl;
}

Pessoa::~Pessoa() {
    Pessoa::decrementa();
    cout << "Destrutor: " << this->nome << endl;
}

string Pessoa::getNome() const {
    return nome;
}

void Pessoa::setNome(string nome) {
    this->nome = nome;
}

int Pessoa::getIdade() const {
    return idade;
}

void Pessoa::setIdade(int idade) {
    this->idade = idade;
}

void Pessoa::imprime() const {
    cout << "Nome: " << nome << endl;
    cout << "Idade: " << idade << endl;
}

int Pessoa::getContador() {
    return Pessoa::contador;
}

void Pessoa::incrementa() {
    Pessoa::contador++;
}

void Pessoa::decrementa() {
    Pessoa::contador--;
}